src/meta/src/hummock/hummock_manager.rs

```diff
@@ -504,10 +494,18 @@ where
                     .map_err(Error::from)
             })
             .await?;
-        let mut compact_status = VarTransaction::new(&mut compaction.compact_status);
+        let mut compact_status = VarTransaction::new(
+            compaction
+                .compaction_statuses
+                .get_mut(&compaction_group_id)
+                .ok_or(Error::InvalidCompactionGroup(compaction_group_id))?,
+        );
         let current_version = self.versioning.read().await.current_version();
-        let compact_task = compact_status
-            .get_compact_task(&current_version.levels, task_id as HummockCompactionTaskId);
+        let compact_task = compact_status.get_compact_task(
```

Comment:
I thought the levels of each group would be split and independent....
So how could we filter the files by group_id?

Reply:
> I thought the levels of each group would be split and independent....

Yes logically they are.

> So how could we filter the files by group_id?

I was thinking about identify a file's group_id
- Either by its key range. It requires both the key range's left and right are valid key from this file.
- Or by a group_id field in its meta.

Let me consider splitting them physically as you mentioned.

Reply:
I think we need a way to identify a file's group anyway. For example, even if we physically split groups' levels, we need to know each new files' groups in `commit_epoch`. 
Therefore, I prefer to not split groups' levels physically. What do you think?

Reply:
Let's split levels by compaction group id in `HummockVersion` in following PR.